src/renderers/WebGLRenderer.js

```diff
@@ -1243,24 +1242,6 @@ class WebGLRenderer {
 
 				currentRenderState = renderStateStack[ renderStateStack.length - 1 ];
 
-				// restore clipping uniforms and viewport
-
-				const renderStateCamera = currentRenderState.state.camera;
-
-				if ( renderStateCamera !== null ) {
-
-					if ( _clippingEnabled === true ) clipping.setGlobalState( _this.clippingPlanes, renderStateCamera );
-
-					const viewport = renderStateCamera.viewport;
-
-					if ( viewport !== undefined ) {
-
-						state.viewport( viewport );
```

Comment:
The issue is that this bit is executed too early when the `Reflector` renders its view. The viewport doesn't match because it is overwritten when the XR render target is set back.